services/blockchain-app-registry/shared/utils/downloadRepository.js

```diff
@@ -151,7 +151,8 @@ const syncWithRemoteRepo = async () => {
 					await downloadFile(result.data.download_url, filePath);
 					logger.debug(`Successfully downloaded: ${file}`);
 
-					await indexMetadataFromFile(filePath);
+					const [, , network, appName, filename] = file.split('/');
+					await indexMetadataFromFile(network, appName, filename);
 					logger.debug('Successfully updated the database with the latest changes');
```

Comment:
Please execute the indexing logic only when the file has a `.json` extension.
```suggestion
					if (file.endsWith('.json')) {
						const [, , network, appName, filename] = file.split('/');
						await indexMetadataFromFile(network, appName, filename);
						logger.debug('Successfully updated the database with the latest changes');
					}
```